pkg/gardenlet/controller/managedseed/managedseed_valueshelper.go

```diff
@@ -299,38 +307,48 @@ func getParentGardenletDeployment(imageVector imagevector.ImageVector, shoot *ga
 		imageTag = version.Get().GitVersion
 	}
 
+	// Get image vector overwrites
+	imageVectorOverwrite, err := getParentImageVectorOverwrite()
+	if err != nil {
+		return nil, err
+	}
+	componentImageVectorOverwrites, err := getParentComponentImageVectorOverwrites()
+	if err != nil {
+		return nil, err
+	}
+
 	// Create and return result
 	return &seedmanagementv1alpha1.GardenletDeployment{
 		Image: &seedmanagementv1alpha1.Image{
 			Repository: &imageRepository,
 			Tag:        &imageTag,
 		},
-		PodAnnotations: getParentPodAnnotations(shoot),
+		PodAnnotations:                 getParentPodAnnotations(shoot),
+		ImageVectorOverwrite:           imageVectorOverwrite,
+		ComponentImageVectorOverwrites: componentImageVectorOverwrites,
 	}, nil
 }
 
-func getParentImageVectorOverwrite() (string, error) {
-	var imageVectorOverwrite string
+func getParentImageVectorOverwrite() (gardencorev1beta1.ImageVector, error) {
 	if overWritePath := os.Getenv(imagevector.OverrideEnv); len(overWritePath) > 0 {
 		data, err := ioutil.ReadFile(overWritePath)
 		if err != nil {
-			return "", err
+			return nil, err
 		}
-		imageVectorOverwrite = string(data)
+		return gardencorev1beta1helper.ReadImageVector(strings.NewReader(string(data)))
```

Comment:
Can you use `os.Open` and use the returned file as a reader instead of reading all bytes and copying them into another reader?

Reply:
There is no copying, `strings.NewReader` just creates a wrapper over the string to allow it to function as a `Reader`. I could do as you suggest, but the only thing that this would save would be some part of the memory to hold `data` (precisely, the difference between the length of `data` and 512). Since our image vectors are rather small, this is negligible. I will also have to introduce `defer os.Close` and proper error handling for everything, so the amount of code will be the same or more. Honestly I prefer the current implementation, but I could of course change it if you insist.

Reply:
I don't insist. But you're right, it shouldn't matter too much.
Resolving this one.